src/logic/nfts/marketplace.ts

```diff
@@ -0,0 +1,255 @@
+              description:
+                (fragment.parcel.data && fragment.parcel.data.description) ||
+                null,
+              x: fragment.parcel.x,
+              y: fragment.parcel.y,
+              estate: fragment.parcel.estate
+                ? {
+                    tokenId: fragment.parcel.estate.tokenId,
+                    name:
+                      (fragment.parcel.estate.data &&
+                        fragment.parcel.estate.data.name) ||
+                      capitalize(NFTCategory.ESTATE),
+                  }
+                : null,
+            }
+          : undefined,
+        estate: fragment.estate
+          ? {
+              description:
+                (fragment.estate.data && fragment.estate.data.description) ||
+                null,
+              size: fragment.estate.size,
+              parcels: fragment.estate.parcels.map(({ x, y }) => ({
+                x: +x,
+                y: +y,
+              })),
+            }
+          : undefined,
+        wearable: fragment.wearable
+          ? {
+              bodyShapes: fragment.wearable.bodyShapes,
+              category: fragment.wearable.category,
+              description: fragment.wearable.description,
+              rarity: fragment.wearable.rarity,
+            }
+          : undefined,
+        ens: fragment.ens ? { subdomain: fragment.ens.subdomain } : undefined,
+      },
+      issuedId: null,
+      itemId: null,
+      category: fragment.category,
+      network: Network.ETHEREUM,
+      chainId: getMarketplaceChainId(),
+      createdAt: +fragment.createdAt * 1000,
+      updatedAt: +fragment.updatedAt * 1000,
+    },
+    order:
+      fragment.activeOrder && !isExpired(fragment.activeOrder.expiresAt)
+        ? fromMarketplaceOrderFragment(fragment.activeOrder)
+        : null,
+  }
+
+  // remove undefined data
+  for (const property of Object.keys(result.nft.data)) {
+    const key = property as keyof typeof result.nft.data
+    if (!result.nft.data[key]) {
+      delete result.nft.data[key]
+    }
+  }
+
+  return result
+}
+
+export function getMarketplaceExtraVariables(options: NFTFilters) {
+  const extraVariables: string[] = []
+  if (options.category) {
+    extraVariables.push('$category: Category')
+  }
+  return extraVariables
+}
+
+export function getMarketplaceExtraWhere(options: NFTFilters) {
+  const extraWhere = ['searchEstateSize_gt: 0', 'searchParcelIsInBounds: true']
+  if (options.category) {
+    extraWhere.push('category: $category')
+  }
+  if (options.isLand) {
+    extraWhere.push('searchIsLand: true')
+  }
+  return extraWhere
+}
+
+export function fromMarketplaceOrderFragment(fragment: OrderFragment) {
+  return fromOrderFragment(fragment, Network.ETHEREUM, getMarketplaceChainId())
+}
+
+export function marketplaceShouldFetch(filters: NFTFilters) {
+  if (
+    (filters.network && filters.network !== Network.ETHEREUM) ||
+    filters.itemId
+  ) {
+    return false
+  } else {
+    return true
```

Comment:
Just because I pointed it out before, but you could just return the (negated) result of the if condition here if you want!